### DISCUSSION TITLE
Question: How do I call an async JavaScript function / can we also generate a post/get api ?

### DISCUSSION THREAD

**Discussion by VegarRingdalAibel** [zauberzeug/nicegui#179] [Q&A]
1:
How do I call an async javascript function ?

This code does not wait for promise to resolve first
```py
async def callJs():
    await ui.run_javascript('myAsyncFunction()')
```

```js

getSomeData = function () {
   return new Promise((resolve)=>{
      setTimeout(()=>{
        resolve(new Date());
      })
  })
};


```

2:

Is there any way to generate POST/GET api in backend, incase the js part want to callback when its done with something ?

**Comment by falkoschindler** [comment]
1. I'm not sure if running async JavaScript functions (and waiting for their result) is possible. Currently, NiceGUI relies on JustPy's `run_javascript` function which uses `eval` on the client side. So far I haven't found a way to eval a promise.

2. Is [`ui.get`](https://nicegui.io/reference#get_decorator) what you're looking for?

**Comment by VegarRingdalAibel** [comment]
1: 
OK, I can prob just build logic around ui.get

2: 
Yes, my bad, I should have been able to find this one. Got lost in my run js issues 🤪
Thx.

**Comment by VegarRingdalAibel** [comment]
Hmm, think I might have closed issue to soon.
Is there a ui.post somewhere?
I want to be able to do something in js, and respond with result when its ready.

**Comment by rodja** [comment]
More comprehensive http support will arrive with [1.0](https://github.com/zauberzeug/nicegui/tree/v1). Until then you may work around this by passing the data as url parameters in a get request. As the [example](https://nicegui.io/reference#get_decorator) is doing with the `id`.

**Comment by me21** [comment]
Guys, I think I enabled async Javascript execution with #180. Please look and test.
In case of topic starter, it would be `ui.run_javascript('return await myAsyncFunction()')`.
